### DISCUSSION TITLE
Adding support for 'meta' commands

### DISCUSSION THREAD

**Discussion by kpfleming** [TD22057/insteon-mqtt#481] [Ideas]
I'd like to be able to have HA automations send commands to change LED brightness, or to change the default 'on' brightness level for a dimmer.

Since the HA 'light' entity doesn't have any concept of such commands, what would be a good approach for this? I'd rather avoid hardcoding MQTT message details in the automations if I can.

**Comment by TD22057** [comment]
IMO MQTT is the approach - the API is the MQTT message.  I suppose in theory the discovery process could be used to create new entities that send/recv those messages for you but the number of options (ane entities) for every insteon device would be enormous and clutter up your home assistant with a ton of stuff you don't want.  And a lot of the ancillary attributes don't have state reporting messages defined so that would be a ton of work to add them.  For example, if you wanted a input_select to control the LED brightness, we'd need to add a new system for reporting that value along with a discovery topic for creating it and a way for users to control whether it get's created or not.  It could be done, but it seems simpler to just send the MQTT message from an automation.
